tests/tools/oesign-engine/test-engine/oesign-test-engine.c

```diff
@@ -0,0 +1,275 @@
+        *size += 1;
+    }
+
+    rc = 0;
+
+done:
+
+    if (rc != 0)
+    {
+        if (data && *data)
+        {
+            free(*data);
+            *data = NULL;
+        }
+
+        if (size)
+            *size = 0;
+    }
+
+    if (is)
+        fclose(is);
+
+    return rc;
+}
+
+
+
+/*
+ * We pass the name of the pem file as the keyid, so that we can actually use the signing key from the openenclave build 
+ * as a signing key for oesign from the test engine. We know the test went well when we can load a test enclave
+ */
+static EVP_PKEY *get_private_key_from_keyid(const char *keyid)
+{
+     uint8_t *pem_data = NULL;
+     size_t  pem_size  = 0;
+     EVP_PKEY* pkey = NULL; 
+
+     const char *path = keyid;  // Do we wanty to directly do that or have an environment variable to direct to a path for pem files?
+     
+     if (! _load_pem_file(path, (void**)&pem_data, &pem_size))
+     {
+         BIO* bio = NULL;
+
+         if (!(bio = BIO_new_mem_buf(pem_data, (int)pem_size)))
+         {
+             goto err;
+         }
+
+         /* Read the key object */
+         if (!(pkey = PEM_read_bio_PrivateKey(bio, &pkey, NULL, NULL)))
+         {
+             goto err;
+         }
+
+         return pkey;
+     }
+
+err:
+     return NULL;
+}
+
+static EVP_PKEY *test_engine_load_privkey(ENGINE *e, const char *keyid,
+                                         UI_METHOD *ui_method,
+                                         void *callback_data)
+{
+
+     (void)ui_method;
+     (void)callback_data;
+     printf("test_engine_load_privkey engine=%p, keyid = %s\n",  e, keyid);
+
+     EVP_PKEY *pkey = get_private_key_from_keyid(keyid);
+     
+     return pkey;
+}
+
+
+static EVP_PKEY *test_engine_load_pubkey(ENGINE *e, const char *keyid,
+                                         UI_METHOD *ui_method,
+                                         void *callback_data)
+{
+     (void)ui_method;
+     (void)callback_data;
+     printf("test_engine_load_pubkey engine=%p, keyid = %s\n",  e, keyid);
+     return test_engine_only_pubkey;
+}
+
+
+static int test_engine_init(ENGINE *e)
+{ 
+    int ret = 0;
+
+    printf("test_engine_init\n");
+
+    test_engine_only_privkey = EVP_PKEY_new();
+    test_engine_only_pubkey  = EVP_PKEY_new();
+    if (!ENGINE_set_load_privkey_function(e, test_engine_load_privkey)){
+        fprintf(stderr, "ENGINE_set_id failed\n");
+        goto done;
+    }
+    if (!ENGINE_set_load_pubkey_function(e, test_engine_load_pubkey)) {
+        fprintf(stderr, "ENGINE_set_id failed\n");
+        goto done;
+    }
+
+    ret = 1;
+
+done:
+    return ret;
+}
+
+static int test_engine_finish(ENGINE *e)
+{ 
+    (void)e;
+    printf("test_engine_finish\n");
+    return 1;
+}
+
+static int test_engine_destroy(ENGINE *e)
+{ 
+    (void)e;
+    printf("test_engine_destroy\n");
+    //2do:test_engine_param_free();
+    EVP_PKEY_free(test_engine_only_privkey);
+    EVP_PKEY_free(test_engine_only_pubkey);
+    return 1;
+}
+
+#if 0
```

Comment:
Why is there code inside a #if 0 ?
Can it be deleted?

Reply:
done